screenshot_windows.go

```diff
@@ -140,11 +142,71 @@ type getMonitorBoundsContext struct {
 func getMonitorBoundsCallback(hMonitor win.HMONITOR, hdcMonitor win.HDC, lprcMonitor *win.RECT, dwData uintptr) uintptr {
 	var ctx *getMonitorBoundsContext
 	ctx = (*getMonitorBoundsContext)(unsafe.Pointer(dwData))
-	if ctx.Count == ctx.Index {
-		ctx.Rect = *lprcMonitor
-		return uintptr(0)
-	} else {
+	if ctx.Count != ctx.Index {
 		ctx.Count = ctx.Count + 1
 		return uintptr(1)
 	}
+
+	if realSize := getMonitorRealSize(hMonitor); realSize != nil {
+		ctx.Rect = *realSize
+	} else {
+		ctx.Rect = *lprcMonitor
+	}
+
+	return uintptr(0)
+}
+
+type _MONITORINFOEX struct {
+	win.MONITORINFO
+	DeviceName [win.CCHDEVICENAME]uint16
 }
+
+// getMonitorRealSize makes a call to GetMonitorInfo
+// to obtain the device name for the monitor handle
+// provided to the method.
+//
+// With the device name, EnumDisplaySettings is called to
+// obtain the current configuration for the monitor, this
+// information includes the real resolution of the monitor
+// rather than the scaled version based on DPI.
+//
+// If either handle calls fail, it will return a nil
+// allowing the calling method to use the bounds information
+// returned by EnumDisplayMonitors which may be affected
+// by DPI.
+func getMonitorRealSize(hMonitor win.HMONITOR) *win.RECT {
+	info := _MONITORINFOEX{}
+	info.CbSize = uint32(unsafe.Sizeof(info))
+
+	ret, _, _ := syscall.Syscall(funcGetMonitorInfo, 2, uintptr(hMonitor), uintptr(unsafe.Pointer(&info)), 0)
+	if ret == 0 {
+		return nil
+	}
+
+	devMode := win.DEVMODE{}
+	devMode.DmSize = uint16(unsafe.Sizeof(devMode))
+
+	if ret, _, _ := syscall.Syscall(funcEnumDisplaySettings, 3, uintptr(unsafe.Pointer(&info.DeviceName[0])), uintptr(0xFFFFFFFF), uintptr(unsafe.Pointer(&devMode))); ret == 0 {
+		return nil
+	}
+
+	// NOTE(lh): Due to the layout of the DEVMODE
+	//           structure, display specific values
+	//           are actually found under printer
+	//           named variables.
+	//
+	//           Due to this, we need to calculate
+	//           the int32 value for the X and Y
+	//           position from the variables in
+	//           the same byte location as the
+	//           wanted variables.
+	positionX := int32(devMode.DmOrientation) | (int32(devMode.DmPaperSize) << 16)
```

Comment:
@LiamHaworth
I think it is more clean to define custom `DEVMODE` struct instead of using `win.DEVMODE` with bitshift.
Could you try this `_DEVMODE` instead of `win.DEVMODE`?

```golang
type _DEVMODE struct {
	DmDeviceName       [win.CCHDEVICENAME]uint16
	DmSpecVersion      uint16
	DmDriverVersion    uint16
	DmSize             uint16
	DmDriverExtra      uint16
	DmFields           uint32
	DmPosition         win.POINT
	padding            [8]byte
	DmColor            int16
	DmDuplex           int16
	DmYResolution      int16
	DmTTOption         int16
	DmCollate          int16
	DmFormName         [win.CCHFORMNAME]uint16
	DmLogPixels        uint16
	DmBitsPerPel       uint32
	DmPelsWidth        uint32
	DmPelsHeight       uint32
	DmDisplayFlags     uint32
	DmDisplayFrequency uint32
	DmICMMethod        uint32
	DmICMIntent        uint32
	DmMediaType        uint32
	DmDitherType       uint32
	DmReserved1        uint32
	DmReserved2        uint32
	DmPanningWidth     uint32
	DmPanningHeight    uint32
}
```

Reply:
Sure can, would of done so originally but wanted to keep the re-engineering of existing structures to a minimum to make the change less messy.

Either way, will make the change, give a test then update the PR.

Reply:
Is something like this acceptable?

```go
type _DEVMODE struct {
	_ [14]byte
	DmPosition         win.POINT
	_ [80]byte
	DmPelsWidth        uint32
	DmPelsHeight       uint32
	_ [40]byte
}
```

Just to cut down on the unnecessary fields, if not I'll use one with all the fields described.

Reply:
Is something like this acceptable?

```go
type _DEVMODE struct {
	_ [14]byte
	DmPosition         win.POINT
	_ [80]byte
	DmPelsWidth        uint32
	DmPelsHeight       uint32
	_ [40]byte
}
```

Just to cut down on the unnecessary fields, if not I'll use one with all the fields described.

Reply:
Yes, of course! 🚀

Reply:
@kbinani I have updated the changes in the pull request (squashed the commits), testing it on my Windows 10 VM works as expected (including handling of scaling). Let me know if there is anything else you would like changed.

Also added a const variable (`_ENUM_CURRENT_SETTINGS`) so there isn't a floating value in one of the syscalls that has no meaning.